src/source/tile_id.js

```diff
@@ -95,6 +95,16 @@ export class OverscaledTileID {
         }
     }
 
+    calculateScaledKey(targetZ: number, wrap: 0 | 1) {
```

Comment:
I remember that `wrap` can also be `-1` in some situations, or `2` if we show tiles in an extreme wide map view.

Reply:
Added comment.
How does it sound https://github.com/mapbox/mapbox-gl-js/commit/46731324bec5dbe294a66ce11b687b1fe68de734#diff-e444e40548c379347d0aa642beffc811R99?
Renamed wrap to withWrap as it has boolean semantics here.

Reply:
it should be fine to use use `wrap: boolean` and `true`/`false` for this

Reply:
> it should be fine to use use `wrap: boolean` and `true`/`false` for this

Prefer to keep 1/0 as they are used in multiplication later. Through ternary or +withWrap would do the same. Documentation should cover it.

Reply:
@ansis Thanks.
After offline discussion,  `withWrap: boolean`used and `+withWrap` in multiplication. Comment updated.
Updated patch pushed.